build.sbt

```diff
@@ -286,6 +245,87 @@ lazy val rspaceBench = (project in file("rspace-bench"))
   .enablePlugins(JmhPlugin)
   .dependsOn(rspace)
 
+lazy val deployment = (project in file("deployment"))
```

Comment:
Are you using the `project in file("<file that does not actually exist>")` hack again?

Really, really don't like this and am not at all happy about the fact that it crept into our project with `rholangCLI`...

Reply:
To add to this, I really don't see the problem with keeping all of this stuff as part of the `node` project.  `node` is, after all, the thing we are deploying...

Reply:
@henrytill You are right. I created the `deployment` project folder on my local machine. But since the folder was empty and `target` is git-ignored it didn't make it into git. Maybe I should add a `README.md` to this folder?

Reply:
No I truly think this stuff belongs in the `node` subproject.  If you wanted to factor out `project/Depolyment.scala` file containing a `deploymentSettings` `Seq` with all the packaging/deployment settings, you could then just add:
```scala
  .settings(deploymentSettings: _*)
```
to the node project definition.  This is not dissimilar to what I did to factor out `CompilerSettings.options`.

But I am definitely not in favor of "dummy" projects like this.

Reply:
There is only one reason why I had to do this:
#### node
```scala
    assemblyExcludedJars in assembly := {
      val cp = (fullClasspath in assembly).value
      cp.filter(a => nettyTcnativeAssemblyExclusions.contains(a.data.getName))
    }
```
This `assembly` configuration is used for creating a fat `jar` for the local system.

#### deployment
```scala
    assemblyExcludedJars in assembly := {
      val cp = (fullClasspath in assembly).value
      cp.filter(a => nettyTcnativeDebianExclusions.contains(a.data.getName))
    },
```
This `assembly` configuration is used for creating a fat `jar` for the `docker` image.

@henrytill I'm afraid there is no way in `sbt` to define compilation path dependent configurations. For example, if someone calls `node/docker` then take this `libraryDependencies` and if someone calls `node/assembly` then take some other `libraryDependencies`. My solution to this problem was to exclude dependencies while packaging because this is under control of the `sbt-native-packager`. This works fine for `deb` and `rpm` packages. However, it doesn't work for `docker` because we are not deploying the artefacts with the `sbt-native-packager` into the image. Instead, we are building a fat `jar` with the `assembly` plug-in and then copy the artefact into the `docker` image. This is the point where things start to go sideways with `assemblyExcludedJars`. I believe, what you did with `CompilerSettings.option` is not related to this problem because you don't have to set different options for `rpm`, `deb` and `docker`.

@rabbitonweb had a different solution for this problem. His suggestion was to introduce an environment variable which would allow us to pick the right set of dependencies (in an `if`-expression). For example `DEPLOYMENT_TYPE=docker sbt node/docker`.

Reply:
As a compromise, I can offer to switch the `deployment` project for some `local` dummy project. Alle the `deployment` settings would go back to the `node` project. The `local` project would serve the single purpose of building a fat jar for the local system.

Reply:
> Instead, we are building a fat jar with the assembly plug-in and then copy the artefact into the docker image

We should not be doing this.  Originally, this was a bad choice by someone who was new to this set tools.  

We should be using `sbt-native-packager` to build the Docker images in much the same way we are building the `deb` and `rpm` packages.